apps/apps.c

```diff
@@ -649,10 +653,12 @@ X509 *load_cert(const char *file, int format, const char *cert_descrip)
         x = d2i_X509_bio(cert, NULL);
     } else if (format == FORMAT_PEM) {
         x = PEM_read_bio_X509_AUX(cert, NULL,
-                                  (pem_password_cb *)password_callback, NULL);
+                                  (pem_password_cb *)password_callback, &cb_data);
     } else if (format == FORMAT_PKCS12) {
-        if (!load_pkcs12(cert, cert_descrip, NULL, NULL, NULL, &x, NULL))
-            goto end;
+        EVP_PKEY *pkey = NULL; /* &pkey is required by PKCS12_parse */
```

Comment:
I had passed in a valid `&pkey` because according to the documentation of `PKCS12_pase()` this is required. Yet it turns out that this is not really the case.
So I've corrected the documentation, removed the needless `pkey` variable, and fixed some style nits.

Reply:
I've kept the declaration of `cb_data` where it was because, as written above,
I believe there can be situations where passing in the password while parsing a cert from a PEM file can make sense.

Reply:
It turned out that still a fix of `PKCS12_parse()` in case `pkey == NULL` and `cert != NULL` was needed, but now everything works fine.